Modules/IxNetRestApiPortMgmt.py

```diff
@@ -381,108 +331,19 @@ def assignPorts(self, portList, forceTakePortOwnership=True, createVports=False,
                   headers={}
             Expecting:   RESPONSE:  SUCCESS
         """
-        # Verify if the portList has duplicates.
-        self.verifyForDuplicatePorts(portList)
+        if createVports :
+            self.createVports(portList)
+        testPorts = []
+        vportList = [vport for vport in self.ixNetObj.Vport.find()]
+        for port in portList:
+            testPorts.append(dict(Arg1=port[0], Arg2=port[1], Arg3=port[2]))
+        self.ixnObj.logInfo('\nAssignPorts: {0}'.format(portList))
 
-        # Verify if there is existing vports. If yes, user either loaded a saved config file or 
-        # the configuration already has vports.
-        # If loading a saved config file and reassigning ports, assign ports to existing vports.
-        response = self.ixnObj.get(self.ixnObj.sessionUrl+'/vport')
+        try :
+            self.ixNetObj.AssignPorts(testPorts, [], vportList, forceTakePortOwnership)
+        except :
+            raise Exception("Failed to Assign ports {} ".format(portList))
```

Comment:
Why not handled rawTraffic, configPortName parameters, those are not required in Restpy?
Missed return statement here, add